test_scripts/ATF_ShutDown_MASTER_RESET.lua

```diff
@@ -0,0 +1,118 @@
+---  1. SDL deletes Apps folder
+---    => file (which had been put to application and stored in application folder) is removed. 
+---  2. SDL clears app info in Policy DB and "resume_app_list" in app_info.dat file 
+---    => SDL will not resume application when the same application registers.
+
+--- Shutdown with reason = "MASTER_RESET".
+commonFunctions:newTestCasesGroup("Check that SDL finish it's working properly by MASTER_RESET")
+function Test:ShutDown_MASTER_RESET()
+  self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications", 
+    { reason = "MASTER_RESET" })
+  EXPECT_NOTIFICATION("OnAppInterfaceUnregistered", { reason = "MASTER_RESET" })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  --ATF is unable to receive SDL notificaion
+  --EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLClose"):Do(function (_,_) 
+  --   print("BasicCommunication.OnSDLClose received")
+  --    StopSDL()
+  --end)
+end
+function Test:Stop_SDL_After_ShutDown_MASTER_RESET()
+  StopSDL()
+end
+
+--- Start SDL again then add mobile session
+function Test:Restart_SDL_and_add_mobile_session()
+    self:runSDL()
+    commonFunctions:waitForSDLStart(self):Do(function()
+      self:initHMI():Do(function()
+      print("Hmi initialized")
+      self:initHMI_onReady():Do(function ()
+        print("Hmi is ready")
+        self:connectMobile():Do(function ()
+          print("Mobile Connected")
+        end)
+      end)
+    end)
+  end)
+end
+
+--- Check SDL will not resume application when the same application registers.
+function Test:Check_Application_Not_Resume_When_Register_Again()
+  local mobile_session = mobile_session.MobileSession(self,self.mobileConnection, nil)
+  local on_rpc_service_started = mobile_session:StartRPC()
+
+  on_rpc_service_started:Do(function()
+    local cid = self.mobileSession:SendRPC("RegisterAppInterface", 
+                                           config.application1.registerAppInterfaceParams)
+    self.mobileSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered", 
+      { application = {appName = config.application1.registerAppInterfaceParams.appName} })
+
+    EXPECT_HMICALL("BasicCommunication.UpdateAppList"):Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, "BasicCommunication.UpdateAppList", "SUCCESS", {})
+    end)
+
+    EXPECT_NOTIFICATION("OnHMIStatus"):ValidIf(function(_,data)
+      return data.payload.hmiLevel == "NONE" and 
+             data.payload.systemContext == "MAIN" and 
+             data.payload.audioStreamingState == "NOT_AUDIBLE"
+    end)
+
+    local cid = self.mobileSession:SendRPC("ListFiles", {})
+    EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS" }) :ValidIf (function(_,data)
+      return not data.payload.filenames
+    end)
+
+    EXPECT_HMICALL("BasicCommunication.ActivateApp"):Times(0)
+    StopSDL()
```

Comment:
@istoimenova please consider. That adding DelayExp(10000) is **very bad pattern**. 
What if SDL will send ActivateApp on 11-s second?  Is this case should be passed?  
The only way to check that SDL didn't sent any RPC is shut down SDL If SDL was finished it's work and up to latest breathe didnt send  this notification, we can guarantee that it wont send it after the process will finish. So waiting for stopping SDL is the only way to check that SDL didn't send something.